src/main/java/com/icthh/xm/tmf/ms/activation/domain/spec/SagaTaskSpec.java

```diff
@@ -7,16 +7,20 @@
 import com.icthh.xm.tmf.ms.activation.domain.SagaTransaction;
 import lombok.Data;
 import lombok.experimental.Accessors;
+import lombok.extern.slf4j.Slf4j;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.function.Consumer;
 import java.util.function.Supplier;
+import java.util.stream.Collectors;
 
+@Slf4j
 @Data
 @JsonInclude(NON_NULL)
 @Accessors(chain = true)
-public class SagaTaskSpec {
+public class SagaTaskSpec implements Cloneable {
```

Comment:
May be usage of constructor to clone could be better option?
- Don’t force us to implement any interface or throw an exception, but we can surely do it if it is required.
- Don’t require any casts.
- Don’t require parent classes to follow any contract or implement anything.
- Allow us to modify final fields.
- Allow us to have complete control over object creation, meaning we can write our initialization logic in it.
I know it is preference question, but anyway approach could be evaluated

Reply:
Could you make a more informative comment?
1. Where you are forced to implement interfaces or throw exceptions?
2. Where it is required? Where you see any casts or their requirements?
3. Where you see any parent classes other than java.lang.Object?
4. Which fields do you mean?
5. Where you see any limitations in object creations?

Reply:
There are good articles with some arguments why:
http://www.javapractices.com/topic/TopicAction.do?Id=71
https://www.javacodegeeks.com/2014/01/which-is-better-option-cloning-or-copy-constructors.html
https://dzone.com/articles/java-cloning-copy-constructor-vs-cloning

I understand it is pretty "hollywar" question, but in modern Java copy constructors and static factory methods considered as better option

Reply:
Very helpful example.
"If somebody does not know how to write cloneable objects, we will get unexpected behavior."
I can answer: if somebody does not know the purpose of the "copy constructor", we will get unexpected behavior.
Even wider: if somebody does not know the java programming language, we will get a lot of unexpected behavior.

Reply:
I've added the Junit test so you don’t forget about cloneable.

Reply:
@AnPogrebnyak, good joke, but what about other arguments? And why not to use copy constructor? I also revised xm-entity, xm-uaa there are no any Cloneable on DTOs and DAOs, so it is not common practice of using Cloneable for this project. 
@amedvedchuk @ishkurko please comment

Reply:
The thing is that we need exactly cloned SagaTaskSpec: it should not have any references to the source, because this object is changed in the task and these changes should not impact on the default behavior of this type of task. Now we change only "next" tasks, but as usual, it can be only begining. The cloneable object is exactly wat we need.

Reply:
I suggest to create package private method `copy` in `SagaTaskSpec`, make class Serializable and use Apache commons util to create deep copy with serialization/deserialization:
```
import org.apache.commons.lang3.SerializationUtils;
...
static SagaTaskSpec copy(SagaTaskSpec source) {
        return SerializationUtils.clone(source);
    }
```

Reply:
Ok, replaced by SerializationUtils.clone